File: reverse.c
#include<stdio.h>
void main()
{
  char str[100],rev[100];
  int i,j,len;
  printf("Enter a string: ");
  gets(str);
  j=0;
  for(i=0;str[i]!='\0';i++);
  len=i;
  for(i=len-1;i>=0;i--){
    rev[j++]=str[i];}
  printf("\n the reversed string is %s",rev);

}

File: reverse.java
import java.util.*;

public class reverse {
  public static void main(String args[]) {
    int len;
    Scanner ob = new Scanner(System.in);
    System.out.println("Enter a string");
    String str = ob.nextLine();
    String rev = " ";
    len = str.length();
    for (int i = len - 1; i >= 0; i--) {
      rev = rev + str.charAt(i);
    }

    System.out.println("The reversed string is " + rev);

  }

}
